pkg/js/validator/validate-store.ts

```diff
@@ -0,0 +1,976 @@
+    if (!userErrors.length) {
+      if (!tuples.map((tuple) => tuple.user).filter((user) => user === checkUser).length) {
+        userErrors.push(undefinedTypeTuple(checkUser, instancePath + "/user"));
+      }
+    }
+  }
+
+  const objectErrors = [];
+
+  if (checkTest.object && isStringValue(checkTest.object)) {
+    const checkObject = checkTest.object;
+    const object = checkObject.split(":")[0];
+
+    // Ensure valid type of object
+    if (!types.includes(object)) {
+      objectErrors.push(invalidTypeUser(object, types, instancePath + "/object"));
+    }
+
+    if (!objectErrors.length) {
+      if (!tuples.map((tuple) => tuple.object).filter((object) => object === checkObject).length) {
+        objectErrors.push(undefinedTypeTuple(checkObject, instancePath + "/object"));
+      }
+    }
+
+    objectErrors.push(...validateAssertionField(model, object, checkTest.assertions, instancePath));
+  }
+
+  const context = checkTest.context;
+  for (const testParam in context) {
+    if (!params.includes(testParam)) {
+      objectErrors.push(unidentifiedTestParam(testParam, instancePath + `/context/${testParam}`));
+    }
+  }
+
+  return [...userErrors, ...objectErrors];
+}
+
+// Validate List Object
+function validateListObject(
+  model: AuthorizationModel,
+  listObjects: ListObjectTest,
+  tuples: TupleKey[],
+  params: string[],
+  instancePath: string,
+) {
+  const errors = [];
+
+  const types = model.type_definitions.map((d) => d.type);
+
+  if (listObjects.user && isStringValue(listObjects.user)) {
+    const listUser = listObjects.user;
+
+    errors.push(...validateUserField(model, types, listUser, instancePath));
+
+    if (!errors.length) {
+      if (!tuples.map((tuple) => tuple.user).filter((user) => user === listUser).length) {
+        errors.push(undefinedTypeTuple(listUser, instancePath + "/user"));
+      }
+    }
+  }
+
+  if (listObjects.type && isStringValue(listObjects.type)) {
+    const listType = listObjects.type;
+
+    // Ensure valid type of object
+    if (!types.includes(listType)) {
+      errors.push(invalidTypeUser(listType, types, instancePath + "/type"));
+    }
+
+    // Validate assertions exist as relations
+    errors.push(...validateAssertionField(model, listType, listObjects.assertions, instancePath));
+  }
+
+  const context = listObjects.context;
+  for (const testParam in context) {
+    if (!params.includes(testParam)) {
+      errors.push(unidentifiedTestParam(testParam, instancePath + `/context/${testParam}`));
+    }
+  }
+  return errors;
+}
+
+// Validate List User
+function validateListUsers(
+  model: AuthorizationModel,
+  listUsers: ListUsersTest,
+  tuples: TupleKey[],
+  params: string[],
+  instancePath: string,
+) {
+  const errors = [];
+
+  const types = model.type_definitions.map((d) => d.type);
+
+  if (listUsers && isStringValue(listUsers.object)) {
+    const listUserObj = listUsers.object;
+
+    const object = listUserObj.split(":")[0];
```

Comment:
```suggestion
    const type = listUserObj.split(":")[0];
```

Reply:
Considering this is used everywhere, can we make this a helper?

Have a tuple parser that converts from a string to the required fields, so that this and similar calls can become: `const { type } = parseObjectString(listUserObj)`


And `parseObjectString` would return `{ type, id }`
And `parseUserString` would return `{ type, id, relation, wildcard }`